bounties/level-2/federation-server.md

```diff
@@ -0,0 +1,109 @@
+# Implement a federation server for stellar addresses
+
+## Award
+| Hunter | Reviewer | Type
+| :- | :- | :-
+| 3200 XLM | 1500 XLM | Capped (2) 
+
+[//]: # (make sure to replace the file-name placeholders `<BOUNTY_FILE_NAME_NO_EXTENSION>`, `<BOUNTY_FILE_NAME_WITH_EXTENSION> and `<LEVEL>` in the next two lines with the respective values)
+📜&nbsp; View [existing submissions](https://github.com/tyvdh/stellar-quest-bounties/issues?q=is%3Aissue+label%3Afederation-server) for this bounty. \
+🔵&nbsp; Start [hunting](https://github.com/tyvdh/stellar-quest-bounties/issues/new?assignees=&labels=&template=begin-the-hunt.yml&link=https://github.com/tyvdh/stellar-quest-bounties/blob/main/bounties/level-2/federation-server.md) this bounty.
+
+
+## Description
+
+We all love our 56-char stellar account-ids. But what about our friends we want to convince to also join stellar-familiy?
+There are a few arguments could stand in the way of adoption:
+
+- Most of the wallets don't have address-book functionality
+- It is mentally challenging to match a stellar-address to an individual ([vanity-addresses](https://lumenaddr.com/faq.html) help a bit – leaving you with only around 50 chars of gobble)
+
+> "Why can't I send you the money via email?"  
+> -— <cite>Your friends</cite>
+
+Well actually they kind of can; enter [federation].
+
+While there are already numerous federation-servers out there – even a [reference implementation] –
+you might not want to trust an external entity with your and your friends' addresses.
+Further, you might want to hook the addresses to an existing community (think discord, github, twitter, …).
+
+
+### What is this task?
+
+Implement a [federation]-server that allows to sign-up (register) with an existing social-media-account provider
+(e.g. discord, github, google, twitter – preferrably one out of the first two to make it easier for the reviewers).
+
+
+### What are the requirements for the bounty hunter?
+
+- You know how to implement and deploy an API
+- You can implement an authentication-flow to allow signing up through a third-party account provider
+- You know how to store and retrieve simple lookups (fed-address/account-id relation)
+
+
+### What are the deliverables?
+
+- A server application that  
+  - allows configuring the domain, that is appended to the federation-addresses
```

Comment:
Just want to make sure I'm reading this bit right. We're allowing the configuration of the domain at the server level, correct? Not a per-user thing? (I'm pretty sure that's impossible anyway?)

Reply:
Exactly. As the server can run anywhere (e.g. cloudflare with their ….dev domains) the domain can not be derived from the DNS or whatever but needs to be configured but not hardcoded in the repo.